WordPress/src/main/java/org/wordpress/android/ui/comments/EditCommentActivity.java

```diff
@@ -177,17 +178,17 @@ private void configureViews() {
         // show error when comment content is empty
         editContent.addTextChangedListener(new TextWatcher() {
             @Override
-            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            public void beforeTextChanged(@NonNull CharSequence s, int start, int count, int after) {
             }
 
             @Override
-            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            public void onTextChanged(@NonNull CharSequence s, int start, int before, int count) {
             }
 
             @Override
-            public void afterTextChanged(Editable s) {
+            public void afterTextChanged(@NonNull Editable s) {
                 boolean hasError = (editContent.getError() != null);
-                boolean hasText = (s != null && s.length() > 0);
```

Comment:
> I agree with your estimation that this probably shouldn't ever be null, but this line makes me wonder...

I think so too... 🤞 

> also, there are a few cases with null guards in the [TextView source](https://android.googlesource.com/platform/frameworks/base/+/jb-mr0-release/core/java/android/widget/TextView.java#3204).. so maybe it's safer to assume nullable?

Can you help me connect the dots of the null guards you shared in the [TextView source](https://android.googlesource.com/platform/frameworks/base/+/jb-mr0-release/core/java/android/widget/TextView.java#3204) and this `@NonNull Editable s`, I honestly fail to find a quick connection, but, I am also a bit rusty with everything `TextView`, `EditText` and `TextWatcher` related, so please forgive me... 😊

Reply:
> Can you help me connect the dots of the null guards you shared

I didn't connect any dots myself, just noted that there were several guards throughout that file where the underlying text (among other things) could be null. That said, it _might_ be the case that it is safe to consider it `@NonNull`, (considering [things like this](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/main/core/java/android/widget/TextView.java#6465), for example). It is unfortunate that this was left unannotated [in the docs](https://developer.android.com/reference/android/text/TextWatcher#afterTextChanged(android.text.Editable)), but my feeling is that it is _probably_ safe. Still, I'd prefer that if we remove a guard, we do it in a separate PR so that we can attach an explicit crash monitoring task, to be proactive in observing the effect in production.